### PR TITLE
C++: Add shared files in `experimental` to `identical-files.json`.

### PR DESCRIPTION
**Pull request by MathiasVP**
When we retargeted https://github.com/github/codeql/pull/10366 to experimental we added a couple of copies of the shared dataflow library. However, I forgot to add those to `identical-files.json`. This PR fixes that.

### CODE REVIEW

**Review by redsun82 — APPROVED** [review]
:shipit:

File: config/identical-files.json [hunk 1]
```diff
@@ -48,6 +53,9 @@
     "cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/tainttracking1/TaintTrackingImpl.qll",
     "cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/tainttracking2/TaintTrackingImpl.qll",
     "cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/tainttracking3/TaintTrackingImpl.qll",
+    "cpp/ql/lib/experimental/semmle/code/cpp/ir/dataflow/internal/tainttracking1/TaintTrackingImpl.qll",
+    "cpp/ql/lib/experimental/semmle/code/cpp/ir/dataflow/internal/tainttracking2/TaintTrackingImpl.qll",
+    "cpp/ql/lib/experimental/semmle/code/cpp/ir/dataflow/internal/tainttracking2/TaintTrackingImpl.qll",
```
Comment (geoffw0) [inline config/identical-files.json:58]:
```suggestion
    "cpp/ql/lib/experimental/semmle/code/cpp/ir/dataflow/internal/tainttracking3/TaintTrackingImpl.qll",
```

**Review by geoffw0 — COMMENTED** [review]
Otherwise LGTM.

File: config/identical-files.json (hunk 1 above)
Comment (MathiasVP) [reply config/identical-files.json]: Oops. Thanks for catching this!

**Review by geoffw0 — APPROVED** [review]

**Pull request merged**